pydantic/_internal/_generate_schema.py

```diff
@@ -1106,6 +1116,72 @@ def _get_or_cache_recursive_ref(self, cls: type[Any]) -> tuple[str, core_schema.
             self.recursion_cache[obj_ref] = core_schema.definition_reference_schema(obj_ref)
             return obj_ref, None
 
+    def _annotated_schema(self, annotated_type: Any) -> core_schema.CoreSchema:
+        """
+        Generate schema for an Annotated type, e.g. `Annotated[int, Field(...)]` or `Annotated[int, Gt(0)]`.
+        """
+        first_arg, *other_args = get_args(annotated_type)
+        return self.apply_annotations(CallbackGetCoreSchemaHandler(self.generate_schema), first_arg, other_args)
+
+    def _get_prepare_pydantic_annotations_for_known_type(self, obj: Any) -> Callable[..., Any] | None:
+        from decimal import Decimal
+
+        from . import _std_types_schema as std_types
+
+        if obj is Decimal:
+            return std_types.decimal_prepare_pydantic_annotations
+
+        return None
+
+    def prepare_annotations(self, source_type: Any, annotations: Iterable[Any]) -> tuple[Any, list[Any]]:
+        prepare = getattr(source_type, '__prepare_pydantic_annotations__', None)
+        annotations = list(annotations)
```

Comment:
looks like this is either unecessary, or should be `tuple(annotations)`.

Reply:
we do need to make sure it's a list because that's what we claim to return. But I moved it down to the return to clarify